ui/number-field.reel/number-field.js

```diff
@@ -1,40 +1,49 @@
 /**
-    @module "montage/ui/native/input-number.reel"
-    @requires montage/ui/component
+    @module "montage/ui/number-field.reel"
     @requires montage/ui/text-input
 */
 
-var TextInput = require("ui/text-input").TextInput;
-
+var TextInput = require("ui/text-input").TextInput,
+    MinMaxConverter = require("core/converter/min-max-converter.js").MinMaxConverter;
+   
 /**
  * Wraps the a &lt;input type="date"> element with binding support for the element's standard attributes.
    @class module:"montage/ui/native/input-number.reel".InputNumber
    @extends module:montage/ui/text-input.TextInput
  */
 var NumberField = exports.NumberField = TextInput.specialize({
-	hasTemplate: {value: false }
+
+    enterDocument:{
+        value: function() {
+            if(this.min || this.max) {
+                this.converter = new MinMaxConverter(this.min, this.max);
```

Comment:
Needs to be modified to not use constructor arguments and instead set .min and .max individually